packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-model-introspection-visitor.test.ts

```diff
@@ -210,6 +210,49 @@ describe('Primary Key Info tests', () => {
     const visitor: AppSyncModelIntrospectionVisitor = getVisitor(schema);
     expect(visitor.generate()).toMatchSnapshot();
   });
+
+  it('should retain order of targetNames and primaryKeyInfo.sortKeyFieldNames', () => {
+    // Data Manager relies on this order matching
+    const schema = /* GraphQL */ `
+      type Enthusiast @model {
+        id: ID! @primaryKey
+        name: String!
+        likes: [Like] @manyToMany(relationName: "EnthusiastLikes")
+      }
+
+      type Like @model {
+        sortKeyFieldThree: String!
+        value: String!
+        name: String! @primaryKey(sortKeyFields: ["sortKeyFieldOne", "sortKeyFieldTwo", "sortKeyFieldThree"])
+        sortKeyFieldOne: String!
+        enthusiasts: [Enthusiast] @manyToMany(relationName: "EnthusiastLikes")
+        sortKeyFieldTwo: String!
+      }
+    `;
+    const result = JSON.parse(getVisitor(schema, { respectPrimaryKeyAttributesOnConnectionField: true }).generate());
+    console.log(result);
+    const { models: { Like, EnthusiastLikes } } = result;
+    console.log(Like.primaryKeyInfo);
+    expect(result).toMatchSnapshot();
```

Comment:
I'd be curious to hear what others think. But, FWIW, in a test like this, I might actually skip the snapshot. The invariants this test cares about are clamped on with other more meaningful assertions.